src/main/java/net/minecraftforge/client/model/generators/BlockStateProvider.java

```diff
@@ -170,6 +170,29 @@ public ModelFile cubeAll(Block block) {
         return models().cubeAll(name(block), blockTexture(block));
     }
 
+    public ModelFile cubeMirroredAll(Block block, String name, ResourceLocation texture) {
+        // Create the standard cube model
+        ModelFile standardModel = models().cubeAll(name, texture);
+
+        // Create the mirrored model by referencing Mojang's mirrored cube template
+        ModelFile mirroredModel = models().withExistingParent(name + "_mirrored", mcLoc("block/cube_mirrored_all"))
+                .texture("all", texture);
+
+        // Generate blockstate with both normal and mirrored models in different rotations
+        getVariantBuilder(block)
+                .forAllStates(state -> new ConfiguredModel[] {
+                        ConfiguredModel.builder().modelFile(standardModel).build(),
+                        ConfiguredModel.builder().modelFile(mirroredModel).build(),
+                        ConfiguredModel.builder().modelFile(standardModel).rotationY(180).build(),
+                        ConfiguredModel.builder().modelFile(mirroredModel).rotationY(180).build()
+                });
+
+        // Return the standard model as a reference
+        return standardModel;
+    }
+
+
+
```

Comment:
These two empty lines after your new method are unnecessary. While it's a nitpick, it's good to maintain consistency within the file.